### ISSUE TITLE
unable to disable proxy for chef provisioners

### ISSUE THREAD

**Issue by warroyo** [tmatilai/vagrant-proxyconf#118]
I am unabel to turn off the proxy for chef. below is the code I am using. it also seems like chef does not honor the no_proxy.

```
   config.proxy.enabled = { chef: false}
   config.proxy.http     = "http://user:pass@myproxy.com:8080"
   config.proxy.https    = "http://user:pass@myproxy.com:8080"
   config.proxy.no_proxy = "localhost,127.0.0.1,10.*"
```

**Comment by otahi** [comment]
Hi @warroyo,
Please clarify your situation more.

**Comment by warroyo** [comment]
sure,

I have added the above lines to my vagrantfile. specifying `chef: false` should prevent the plugin from adding the proxy lines to my solo/client.rb file. this is not the case, I am still seeing the "Configuring proxy for Chef provisioners..." line in the vagrant provision and the client/solo.rb contain proxy settings.

so the problem is that specifying `chef:false` is not actuallly skipping the chef proxy step.

**Comment by codylane** [comment]
I see that this is a rather old issue but I believe that i've addressed this and hope to have a PR put in in the next few days.  But yeah, this caught me off guard too and it was painstaking long to re-create the vagrant environment when all I really needed was to just disable the proxy settings under certain situations.  

Anyway, I would welcome the feedback once I get this PR in.  I'll link this issue to the PR once I have time to create the PR.

**Comment by codylane** [comment]
This should be resolved in the 2.0.0 release. Please confirm.

I'll close out this issue on Jan 11, 2019 if I don't hear from you by then.  Thanks.

**Comment by codylane** [comment]
Closing out issue, please feel free to reopen if this is still an issue.  

Thanks again for submitting this issue.
